tests/keras/engine/test_training.py

```diff
@@ -33,6 +34,23 @@ def __getitem__(self, idx):
 
 
 @keras_test
+def test_standardize_input_data():
+    a_np = np.random.random((4, 3))
+    a_shape = a_np.shape
+    a_name = 'input_a'
+    p = K.placeholder(a_shape)
+    a = Input(shape=a_shape, name='a_name')
+    x = _standardize_input_data(
+        [a_np], [a_name], [a_shape],
+        check_batch_axis=False,
+        exception_prefix='input')
+    x = _standardize_input_data(
+        [None], [a_name], [a_shape],
+        exception_prefix='input',
+        skip_standardizing=[p])
```

Comment:
If you replace `skip_standardizing=[p]` (line 50) with `skip_standardizing=['p']` you get the same behaviour. This is my main objection here. Why specify the variable to skip when the only thing that counts is the length of the skip list?